### DISCUSSION TITLE
Issues Deep Linking to Search Verticals

### DISCUSSION THREAD

**Discussion by CamHowarthInForm** [microsoft-search/pnp-modern-search#2857] [Q&A]
**Version used**
4.8.0.0

**Describe the bug**
I am trying to deep link to search verticals using URL query strings, and I am encountering two possible bugs.

1. links with `href="?v=verticalName"` need to be clicked twice in order to be directed to the vertical.
> The first click adds the query string to the url in the browser's address bar, but does not load the page
> The second click triggers the browser to navigate to the vertical
> I am able to make this work with one click by adding the following attributes to the <a> tag,
> `<a data-interception="off" rel="noopener noreferrer" href=...` I'm unclear why this works.

2. Once the query string for 'vertical A' has been added to the url, clicking on 'vertical B' navigates to 'vertical B' but the query string for 'vertical A' remains in the url.
> I have tried to get around this by treating each vertical as a hyperlink to the search page with its relevant query string. However this approach also encounters issue 1. Each vertical needs to be clicked twice in order to navigate successfully.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Set up a SharePoint page with Search Verticals, Search Results
2. Modify the Search Results webpart to use a custom HTML layout, where each result item contains a link to a vertical. Do not add the `data-interception` and `rel` attributes to the link tag at this stage.
3. In the vertical settings, ensure the vertical query string parameter matches the one used in your link hrefs. I have stuck with the default `v`
4. Publish the page, perform a search, and click the vertical link. It should take two clicks for your browser to navigate to the vertical.

**Expected behavior**
I expect any link that adds query strings to a url to trigger my browser to navigate to that URL after one click.
I expect that clicking on search verticals should clear any irrelevant query strings from my browser address bar.

**Desktop (please complete the following information):**
 - Browser: Tested on Chrome, Edge & Firefox. Same behaviour on all three.

Thanks for your help!

**Comment by wobba** [comment]
Take a look at https://learn.microsoft.com/en-us/sharepoint/dev/spfx/hyperlinking which is likely your issue on your links. As data interception should be done on links to ensure clicks work as you want and are not intercepted by the SPFx URL router.
